app/controllers/authenticate_controller.rb

```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
+require 'wannabe_bool'
```

Comment:
what does this gem give us?

Reply:
convert string to bool.
"true" -> true
"yes" -> true
"1" -> true

"false"  -> false
"no" -> false
"0" -> false


<string which has no boolean meaning> -> false

In ruby there is no `to_boolean` to String and we need to convert the `encoded` parameter from the request (a string) to bool

Reply:
why can't we have:
"true" -> true
anything else (including nil) -> false

Reply:
This library does the same thing. including nil. But it converts to true other logical true strings such as yes... 

I think that it is a good place to use this 3 party in order to avoid:
1. `params[:encoded] == "true"` using true as a literal string seems wrong to me.
2. extending String:
``` 
Class String
    def true?
        return self == "true"
 ```

The 3 party extends the `.to_bool` method which is very clear and also take care of nil.

Reply:
I'll second that:
1. We should only add new dependencies (gems) when the value is clear and obvious, and after it's been discussed by the team as a whole. Dependencies come with future maintenance costs, that I think in this case outweighs the value of boolean parsing. [See here for more info](https://github.com/conjurinc/docs/blob/master/reference/debt_free_rules.md#9-new-tools-require-documented-public-discussion-avoid-unnecessary-tools).

2. Keeping the interface simple is preferred than accepting all options for boolean values ([see the discussion here](https://github.com/cyberark/conjur-service-broker/pull/76)).

Reply:
I'm going to third this just emphasize the point, because I think it's so critical.  I also want to spread this information among team members -- not just the information itself but the reasons for it.

When I wrote the ["no new tools without public disucssion"](https://github.com/conjurinc/docs/blob/master/reference/debt_free_rules.md#9-new-tools-require-documented-public-discussion-avoid-unnecessary-tools) rule that Micah referenced, ruby gems were one of the things I had in mind. 

We want a very high bar for adding new gem dependencies.  In the majority of cases, the better decision -- for clarity of code, long term maintenance, and avoiding security vulns -- will be to use vanilla ruby rather than a gem.

In this particular case, there is especially no reason for it because the logic we want can be written in a line or a few lines.

> extending String:

As a separate point, this is something that should _never_ be done, under any circumstances.  Monkey patching built-in classes is extremely brittle, and just asking for trouble.  The standard library classes represent an important invariant that every experienced ruby programmer can be assumed to understand.  Invisibly changing their behavior breaks that invariant.

If you do want a special String or Array, etc object with new methods, the preferred method is to wrap the object in a decorator that adds them.  This makes what you're doing explicit.